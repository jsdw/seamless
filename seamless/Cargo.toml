[package]
name = "seamless"
version = "0.2.0"
authors = ["James Wilson <james@jsdw.me>"]
edition = "2018"
license = "MIT"
readme = "README.md"
repository = "https://github.com/jsdw/seamless"
categories = ["web-programming", "web-programming::http-server", "asynchronous"]
keywords = ["api", "http", "rpc", "json"]
description = """
An opinionated library to easily create RPC style JSON APIs
"""

[features]
default = ["uuid", "chrono"]

[dependencies]
async-trait = "0.1.36"
serde = { version = "1", features = ["derive"] }
serde_json = "1"
http = "0.2.1"
seamless_macros = { version = "0.2.0", path = "../seamless-macros" }

# impls for these if we want them:
uuid = { version = "0.8", optional = true, features = ["serde"] }
chrono = { version = "0.4.11", optional = true, features = ["serde"] }

[dev-dependencies]
trybuild = "1.0"
pretty_assertions = "0.6.1"
thiserror = "1.0.23"
tokio = { version = "1.1.0", features = ["rt", "rt-multi-thread"] }
