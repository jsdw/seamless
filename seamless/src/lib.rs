pub mod router;
pub mod error;
pub mod body;

pub use seamless_macros::*;

pub use async_trait::async_trait;

pub use router::{
    Api,
    Context,
    Method,
    RouteError,
    RouteInfo,
    Json,
    Binary
};

pub use body::{
    ApiBody,
    ApiBodyType,
    Type,
};

pub use error::{
    ApiError,
    IntoApiError
};