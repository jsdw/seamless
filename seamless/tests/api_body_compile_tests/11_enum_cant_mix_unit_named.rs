use seamless::ApiBody;

#[ApiBody]
enum Lark {
    UnitField,
    NamedFields {}
}

fn main () {

}