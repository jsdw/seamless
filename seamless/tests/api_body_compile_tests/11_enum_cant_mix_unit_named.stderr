error: Unit enum fields can't be mixed with named fields
 --> $DIR/11_enum_cant_mix_unit_named.rs:4:6
  |
4 | enum Lark {
  |      ^^^^
